### PR TITLE
Prototype WDL execution with MiniWDL

### LINKED ISSUE #3468: Prototype miniWDL support
┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-810)
┆Issue Number: TOIL-810
┆Link To Issue: https://ucsc-cgl.atlassian.net/browse/TOIL-810

### PR DESCRIPTION
**Pull request by adamnovak**
This will fix #3468. This adds a prototype WDL runner that uses MiniWDL much the way `toil-cwl-runner` uses `cwltool`.

Mostly it doesn't work, but it works enough to run the MiniWDL self-test workflow, which is now included in the Toil repo.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * New prototype WDL runner `python -m toil.wdl.wdltoil` using MiniWDL

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Comment by DailyDreaming** [comment]
@adamnovak Is this ready for review, or wait?

**Comment by adamnovak** [comment]
This is ready to review.

**Comment by adamnovak** [comment]
CI seemed to have a critical existence failure here; all the jobs in the second stage failed without really properly timing out or complaining about any errors.

File: src/toil/job.py [hunk 1]
```diff
@@ -3263,6 +3268,39 @@ def _resolve(cls, jobStoreLocator, jobStoreFileID):
...
+Promised = Union[Promise, T]
+
+def unwrap(p: Promised[T]) -> T:
+    """
+    Function for ensuring you actually have a promised value, and not just a promise.
+    Mostly useful for satisfying type-checking.
+
+    The "unwrap" terminology is borrowed from Rust.
+    """
+    assert not isinstance(p, Promise)
```
Comment (mr-c) [inline src/toil/job.py:3291]:
```suggestion
    if not isinstance(p, Promise):
        raise AssertionError(f"{p!r} is not a Promise, but a {type(p}).")
```
In many configurations, `assert` statements are skipped

**Review by DailyDreaming — APPROVED** [review]
Overall, this looks great.  Only minor comments and I'd be fine with merging this as is.

Also agree with @mr-c that `Exception` is generally preferable to `assert`.

**Pull request merged**